ngraph/src/ngraph/op/prior_box.cpp

```diff
@@ -145,3 +146,54 @@ bool op::PriorBox::visit_attributes(AttributeVisitor& visitor)
     visitor.on_attribute("attrs.scale_all_sizes", m_attrs.scale_all_sizes);
     return true;
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    bool evaluate(const HostTensorPtr& arg0,
+                  const HostTensorPtr& arg1,
+                  const HostTensorPtr& out,
+                  op::PriorBoxAttrs attrs)
+    {
+        runtime::reference::prior_box(arg0->get_data_ptr<ET>(),
```

Comment:
Just for my clarification, If we are striving to have all reference implementation in the `evaluate` interface, does it really make sense to keep "ngraph/runtime/reference/prior_box.hpp"? Why not move the code of reference implementation right here instead of wrapping one method to another? 

As far as I understand, the only place where `runtime::reference`  is used it is interpreter backend, which has a priority of calling of `evaluate` method, and only if the call was unsuccessful then it should generate calls basing on pure references as a fallback case.

So I'm just to get rid of the mess (for myself first of all) with and approaches how we implement references.

Reply:
As far as I know, you're right, we use reference implementations for ConstantFolding (`evaluate `method) and with an interpreter backend. But to be honest, I am not aware of the architectural idea of this.  For me now it is a historical fact that we use separate files for the reference implementations.  Maybe @GlebKazantaev knows in more detail